nixos/modules/services/backup/burpServer.nix

```diff
@@ -0,0 +1,83 @@
+{ config, lib, networking, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.burpServer;
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.burpServer = {
+
+      enable = mkOption {
+        default = false;
+        description = "Whether to enable burp.";
+      };
+
+      user = mkOption {
+        default = "burp";
+        description = "The user to run burp as";
+      };
+
+      group = mkOption {
+        default = cfg.user;
+        description = "The group to run burp as (default same as user)";
+      };
+
+      configFile = mkOption {
+        default = "/etc/burp/burp.conf";
+        description = "Path to the burp config file. You need to provide a fully working configuration file and setup all paths yourself.";
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.burp;
+        example = literalExample "pkgs.burp";
+        description = ''burp package to use.'';
+      };
+
+    };
+
+  };
+
+  ###### implementation
+
+  config = mkIf config.services.burpServer.enable {
+
+    users.extraUsers.burp = {
```

Comment:
Typically, daemon users are only added if they match some pre-defined default; changing existing users is usually not a good idea (what if, for some perverse reason, the user sets their login as the burp user?)

I suggest either guarding this or making it non-configurable. 

Also, though perhaps a matter of taste, I believe it's not necessary to make something configurable just because you can, but perhaps there are legitimate reasons for wanting to use something other than the default. This applies to the `package` option as well.